liquibase-integration-tests/src/test/groovy/liquibase/command/core/GenerateChangeLogEmptyIntegrationTest.groovy

```diff
@@ -0,0 +1,99 @@
+package liquibase.command.core
+
+import liquibase.Scope
+import liquibase.command.CommandScope
+import liquibase.command.core.helpers.DbUrlConnectionArgumentsCommandStep
+import liquibase.command.core.helpers.DiffOutputControlCommandStep
+import liquibase.command.core.helpers.PreCompareCommandStep
+import liquibase.command.util.CommandUtil
+import liquibase.extension.testing.testsystem.DatabaseTestSystem
+import liquibase.extension.testing.testsystem.TestSystemFactory
+import liquibase.extension.testing.testsystem.spock.LiquibaseIntegrationTest
+import liquibase.util.FileUtil
+
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+@LiquibaseIntegrationTest
+class GenerateChangeLogEmptyIntegrationTest extends Specification {
```

Comment:
For tests I do have two comments:

1. Let's add a new method for calling `generate-changelog` by specifying `diffTypes` and `includeSchema` args in the `CommandUtil` test class:

```groovy
 private void runGenerateChangelog(DatabaseTestSystem db, String outputFile, String diffTypes) throws CommandExecutionException {
        CommandScope commandScope = new CommandScope(GenerateChangelogCommandStep.COMMAND_NAME)
        commandScope.addArgumentValue(DbUrlConnectionArgumentsCommandStep.URL_ARG, db.getConnectionUrl())
        commandScope.addArgumentValue(DbUrlConnectionArgumentsCommandStep.USERNAME_ARG, db.getUsername())
        commandScope.addArgumentValue(DbUrlConnectionArgumentsCommandStep.PASSWORD_ARG, db.getPassword())
        commandScope.addArgumentValue(GenerateChangelogCommandStep.OVERWRITE_OUTPUT_FILE_ARG, true)
        commandScope.addArgumentValue(GenerateChangelogCommandStep.CHANGELOG_FILE_ARG, outputFile)
        commandScope.addArgumentValue(PreCompareCommandStep.DIFF_TYPES_ARG, diffTypes)
        commandScope.addArgumentValue(DiffOutputControlCommandStep.INCLUDE_SCHEMA_ARG, true)
        OutputStream outputStream = new ByteArrayOutputStream()
        commandScope.setOutput(outputStream)
        commandScope.execute()
    }
```
2.  `CommandUtil.runDropAll(db)` from `cleanup` section is not needed because it's already being called from `LiquibaseIntegrationMethodInterceptor`.